krypto/src/commonMain/kotlin/com/soywiz/krypto/encoding/Base64.kt

```diff
@@ -1,42 +1,58 @@
 package com.soywiz.krypto.encoding
 
 object Base64 {
-    private val TABLE = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/="
-    private val DECODE = IntArray(0x100).apply {
+    private const val TABLE = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/="
+    private const val TABLE_SAFE = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_="
+
+    private val decoder = IntArray(0x100).apply {
         for (n in 0..255) this[n] = -1
-        for (n in 0 until TABLE.length) {
+        for (n in TABLE.indices) {
             this[TABLE[n].toInt()] = n
         }
     }
 
+    private val decoderUrlSafe = IntArray(0x100).apply {
+        for (n in 0..255) this[n] = -1
+        for (n in TABLE_SAFE.indices) {
+            this[TABLE_SAFE[n].toInt()] = n
+        }
+    }
+
     operator fun invoke(v: String) = decodeIgnoringSpaces(v)
     operator fun invoke(v: ByteArray) = encode(v)
 
-    fun decode(str: String): ByteArray {
+    fun decode(str: String, isUrlSafe: Boolean = false): ByteArray {
```

Comment:
I'm used to see the `is` as "asking", usually for a boolean getter, and here I would use urlSafe or at least doUrlSafe. Do you have a different criteria there?
Alternatively, why don't just create a base private decode method receiving the right table as parameter, and then two methods one for normal and another for url-safe? That way no ifs would be required for selecting the table

Reply:
Done